### PR TITLE
Factor out less common code-path into own method to allow inlining.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

During benchmarks two methods showed up as "hot method too big". We can easily make these smaller by factor out some less common code-path to an extra method and so allow inlining.

Modifications:

Factor out less common code path to an extra method.

Result:

Hot methods can be inlined.

### CODE REVIEW

File: transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java [hunk 1]
```diff
@@ -434,21 +434,9 @@ private void clearNioBuffers() {
...
-                        for (int i = 0; i < nioBufs.length && nioBufferCount < maxCount; ++i) {
-                            ByteBuffer nioBuf = nioBufs[i];
-                            if (nioBuf == null) {
-                                break;
-                            } else if (!nioBuf.hasRemaining()) {
-                                continue;
-                            }
-                            nioBuffers[nioBufferCount++] = nioBuf;
-                        }
+                        // The code resist in an extra method to ensure the method is not too big to inline as this
```
Comment (jvirtanen) [inline transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java]: "The code *exists* in..."?

File: transport/src/main/java/io/netty/channel/nio/NioEventLoop.java [hunk 2]
```diff
@@ -793,17 +793,9 @@ private void select(boolean oldWakenUp) throws IOException {
...
-                    logger.warn(
-                            "Selector.select() returned prematurely {} times in a row; rebuilding Selector {}.",
-                            selectCnt, selector);
-
-                    rebuildSelector();
-                    selector = this.selector;
-
-                    // Select again to populate selectedKeys.
-                    selector.selectNow();
+                    // The code resist in an extra method to ensure the method is not too big to inline as this
```
Comment (jvirtanen) [inline transport/src/main/java/io/netty/channel/nio/NioEventLoop.java]: "The code _exists_ in ..."?

File: transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java (hunk 1 above)
Comment (normanmaurer) [reply transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java]: fat fingers :D thanks

**Comment by normanmaurer** [comment]
@jvirtanen fixed

File: transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java [hunk 3]
```diff
@@ -463,6 +451,25 @@ private void clearNioBuffers() {
         return nioBuffers;
     }
 
+    private int nioBuffers(Entry entry, ByteBuf buf, ByteBuffer[] nioBuffers, int nioBufferCount, int maxCount) {
```
Comment (njhill) [inline transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java]: static?

**Comment by normanmaurer** [comment]
Alright merging this as it just moves code around :)

**Pull request merged**